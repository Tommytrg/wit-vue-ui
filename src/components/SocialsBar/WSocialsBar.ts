export const SOCIAL_URLS: Record<string, string> = {
  discord: 'https://discord.gg/witnet',
  github: 'https://github.com/witnet',
  telegram: 'https://t.me/witnetio',
  x: 'https://twitter.com/witnet_io'
}
