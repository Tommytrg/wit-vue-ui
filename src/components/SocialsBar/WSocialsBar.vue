<template>
  <div
    class="opacity-60 hover:opacity-100 grid grid-rows-1 grid-flow-row content-center gap-y-md w-fit justify-items-center"
  >
    <IconExternalLink :url="SOCIAL_URLS.x" class="link">
      <WIcon :name="IconName.X" class="icon" />
    </IconExternalLink>
    <IconExternalLink :url="SOCIAL_URLS.telegram" class="link">
      <WIcon :name="IconName.Telegram" class="icon" />
    </IconExternalLink>
    <IconExternalLink :url="SOCIAL_URLS.discord" class="link">
      <WIcon :name="IconName.Discord" class="icon" />
    </IconExternalLink>
    <IconExternalLink :url="SOCIAL_URLS.github" class="link">
      <WIcon :name="IconName.Github" class="icon" />
    </IconExternalLink>

    <!-- TODO(#413): set width to 40px to avoid wrong alignment -->
    <div class="flex mt-3xl w-xl">
      <p class="text-sm font-light text-white-400 [writing-mode:vertical-lr] -rotate-180 subtitle">
        Enjoy Smart, Neutral,
      </p>
      <p class="text-sm font-light text-white-400 [writing-mode:vertical-lr] -rotate-180 subtitle">
        and Secure Data
      </p>
    </div>
  </div>
</template>

<script setup lang="ts">
import WIcon from '../icon/WIcon.vue'
import { IconName } from '../icon/WIcon'
import IconExternalLink from './IconExternalLink.vue'
import { SOCIAL_URLS } from './WSocialsBar'
</script>

<style lang="scss" scoped>
.link {
  @apply justify-self-end
}
.icon {
  @apply w-[18px] h-auto
}
</style>