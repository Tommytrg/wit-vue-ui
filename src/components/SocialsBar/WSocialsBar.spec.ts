import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import WSocialsBar from './WSocialsBar.vue'

describe('SocialsBar', () => {
  it('renders properly', () => {
    const wrapper = mount(WSocialsBar)

    expect(wrapper.element).toMatchSnapshot()
  })
})
