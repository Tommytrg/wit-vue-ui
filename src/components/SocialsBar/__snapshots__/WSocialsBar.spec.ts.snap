// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SocialsBar > renders properly 1`] = `
<div
  class="opacity-60 hover:opacity-100 grid grid-rows-1 grid-flow-row content-center gap-y-md w-fit justify-items-center"
  data-v-5ed1f484=""
>
  <a
    aria-label="https://twitter.com/witnet_io"
    class="link"
    data-v-5ed1f484=""
    href="https://twitter.com/witnet_io"
    target="_blank"
  >
    
    <svg
      class="icon"
      data-v-5ed1f484=""
      data-v-c15836e8=""
      fill="none"
      height="396"
      viewBox="0 0 396 396"
      width="396"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="social-icon"
        d="M301.026 37.125h54.582l-119.246 136.29 140.283 185.46h-109.84l-86.031-112.48-98.439 112.48H27.72l127.545-145.777L20.691 37.125H133.32l77.764 102.812zm-19.157 289.08h30.245L116.886 68.079H84.431z"
        fill="#CCC"
      />
    </svg>
    
  </a>
  <a
    aria-label="https://t.me/witnetio"
    class="link"
    data-v-5ed1f484=""
    href="https://t.me/witnetio"
    target="_blank"
  >
    
    <svg
      class="icon"
      data-v-5ed1f484=""
      data-v-c15836e8=""
      fill="none"
      height="20"
      viewBox="0 0 24 20"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="social-icon"
        clip-rule="evenodd"
        d="M18.384 19.779a1.19 1.19 0 0 0 1.107.145 1.16 1.16 0 0 0 .724-.84C21.084 15 23.192 4.663 23.983.948a.78.78 0 0 0-.26-.758.8.8 0 0 0-.797-.14C18.733 1.602 5.82 6.447.542 8.4A.83.83 0 0 0 0 9.199c.012.354.25.661.593.764 2.367.708 5.474 1.693 5.474 1.693s1.452 4.385 2.209 6.615c.095.28.314.5.603.576a.87.87 0 0 0 .81-.207l3.097-2.923s3.572 2.619 5.598 4.062m-11.01-8.677 1.679 5.538.373-3.507L19.61 3.947a.277.277 0 0 0 .033-.377.284.284 0 0 0-.376-.064z"
        fill="#CCC"
        fill-rule="evenodd"
      />
    </svg>
    
  </a>
  <a
    aria-label="https://discord.gg/witnet"
    class="link"
    data-v-5ed1f484=""
    href="https://discord.gg/witnet"
    target="_blank"
  >
    
    <svg
      class="icon"
      data-v-5ed1f484=""
      data-v-c15836e8=""
      fill="none"
      height="24"
      viewBox="0 0 21 24"
      width="21"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="social-icon"
        clip-rule="evenodd"
        d="M18.54 0A2.466 2.466 0 0 1 21 2.472V24l-2.58-2.28-1.452-1.344-1.536-1.428.636 2.22H2.46A2.466 2.466 0 0 1 0 18.696V2.472A2.466 2.466 0 0 1 2.46 0zm-4.632 15.672c2.652-.084 3.672-1.824 3.672-1.824 0-3.864-1.728-6.996-1.728-6.996-1.728-1.296-3.372-1.26-3.372-1.26l-.168.192c2.04.624 2.988 1.524 2.988 1.524A9.8 9.8 0 0 0 9.264 6.18l-.204.024c-.42.036-1.44.192-2.724.756-.444.204-.708.348-.708.348s.996-.948 3.156-1.572l-.12-.144s-1.644-.036-3.372 1.26c0 0-1.728 3.132-1.728 6.996 0 0 1.008 1.74 3.66 1.824 0 0 .444-.54.804-.996-1.524-.456-2.1-1.416-2.1-1.416l.336.204.048.036.047.027.014.006.047.027c.3.168.6.3.876.408.492.192 1.08.384 1.764.516.9.168 1.956.228 3.108.012a8 8 0 0 0 1.74-.516c.42-.156.888-.384 1.38-.708 0 0-.6.984-2.172 1.428.36.456.792.972.792.972m-5.58-5.604c-.684 0-1.224.6-1.224 1.332s.552 1.332 1.224 1.332c.684 0 1.224-.6 1.224-1.332.012-.732-.54-1.332-1.224-1.332m4.38 0c-.684 0-1.224.6-1.224 1.332s.552 1.332 1.224 1.332c.684 0 1.224-.6 1.224-1.332s-.54-1.332-1.224-1.332"
        fill="#ccc"
        fill-rule="evenodd"
      />
    </svg>
    
  </a>
  <a
    aria-label="https://github.com/witnet"
    class="link"
    data-v-5ed1f484=""
    href="https://github.com/witnet"
    target="_blank"
  >
    
    <svg
      class="icon"
      data-v-5ed1f484=""
      data-v-c15836e8=""
      fill="none"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="social-icon"
        clip-rule="evenodd"
        d="M16 0C7.16 0 0 7.16 0 16c0 7.08 4.58 13.06 10.94 15.18.8.14 1.1-.34 1.1-.76 0-.38-.02-1.64-.02-2.98-4.02.74-5.06-.98-5.38-1.88-.18-.46-.96-1.88-1.64-2.26-.56-.3-1.36-1.04-.02-1.06 1.26-.02 2.16 1.16 2.46 1.64 1.44 2.42 3.74 1.74 4.66 1.32.14-1.04.56-1.74 1.02-2.14-3.56-.4-7.28-1.78-7.28-7.9 0-1.74.62-3.18 1.64-4.3-.16-.4-.72-2.04.16-4.24 0 0 1.34-.42 4.4 1.64 1.28-.36 2.64-.54 4-.54s2.72.18 4 .54c3.06-2.08 4.4-1.64 4.4-1.64.88 2.2.32 3.84.16 4.24 1.02 1.12 1.64 2.54 1.64 4.3 0 6.14-3.74 7.5-7.3 7.9.58.5 1.08 1.46 1.08 2.96 0 2.14-.02 3.86-.02 4.4 0 .42.3.92 1.1.76C27.42 29.06 32 23.06 32 16c0-8.84-7.16-16-16-16"
        fill="#ccc"
        fill-rule="evenodd"
      />
    </svg>
    
  </a>
  <!-- TODO(#413): set width to 40px to avoid wrong alignment -->
  <div
    class="flex mt-3xl w-xl"
    data-v-5ed1f484=""
  >
    <p
      class="text-sm font-light text-white-400 [writing-mode:vertical-lr] -rotate-180 subtitle"
      data-v-5ed1f484=""
    >
       Enjoy Smart, Neutral, 
    </p>
    <p
      class="text-sm font-light text-white-400 [writing-mode:vertical-lr] -rotate-180 subtitle"
      data-v-5ed1f484=""
    >
       and Secure Data 
    </p>
  </div>
</div>
`;
